### PR TITLE
refactor: move classes to a more sensible package hierarchy

### LINKED ISSUE #96: Connector moves and renames
Renames to allow for other endpoints in future (we'll get the cloudant part from the package name):
* `CloudantSourceConnector` -> `ChangesSourceConnector`
* `CloudantSinkConnector` -> `ChangesSinkConnector`

Put the "public" content in these packages
`com.ibm.cloud.cloudant.kafka.connect.connectors` (for the *SourceConnector and *SinkConnector classes)
`com.ibm.cloud.cloudant.kafka.connect.transforms` (for the transforms)
`com.ibm.cloud.cloudant.kafka.connect.transforms.predicates` (for the predicates)

Put the other content in a sensible set of sub-packages in 
`com.ibm.cloud.cloudant.kafka.connect.internal`

### PR DESCRIPTION
**Pull request by mojito317**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [x] Added tests for code changes _or_ test/build only changes
- [x] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [x] Completed the PR template below:

## Description
Fixes #96

## Approach
Moves/renames:
```
src/main/java/com/ibm/cloud/cloudant/kafka/
{connect => caching}/CachedClientManager.java
{schema => mappers}/ConnectRecordMapper.java
{schema => mappers}/DocumentToSourceRecord.java
{connect => tasks}/ConnectorConfig.java
{connect => tasks}/SinkConnectorConfig.java
{connect => tasks}/SinkTask.java
{connect => tasks}/SourceChangesConnectorConfig.java
{connect => tasks}/SourceChangesTask.java
{connect => validators}/ConfigValidator.java
{connect => validators}/ListRecommender.java
{connect => validators}/UrlValidator.java
{common => utils}/CloudantConst.java
{common => utils}/InterfaceConst.java
{common => }/utils/JavaCloudantUtil.java
{common => utils}/MessageKey.java
{common => }/utils/ResourceBundleUtil.java
{connect => }/transforms/ArrayFlatten.java
{connect => }/transforms/MapToStruct.java
{connect => }/transforms/predicates/IsDesignDocument.java

src/test/java/com/ibm/cloud/cloudant/kafka/
{connect => caching}/ClientManagerUtils.java
{schema => mappers}/ConnectRecordMapperTests.java
{schema => mappers}/DocumentHelpers.java
{schema => mappers}/DocumentToSourceRecordTest.java
{connect => tasks}/SinkErrorTest.java
{connect => tasks}/SinkTaskTest.java
{connect => tasks}/SourceChangesAndCloudantSinkTest.java
{connect => tasks}/SourceChangesTaskTest.java
{connect => tasks}/TombstoneTest.java
{connect => }/utils/CloudantDbUtils.java
{connect => }/utils/ConnectorUtils.java
{connect => }/utils/ServiceCallUtils.java
{connect => }/transforms/ArrayFlattenTest.java
{connect => }/transforms/MapToStructTest.java
{connect => }/transforms/predicates/IsDesignDocumentTest.java
{ => validators}/ValidationTest.java
```

## Schema & API Changes

- "No change"

## Security and Privacy

- "No change"

## Testing

- N/A build or packaging only changes

## Monitoring and Logging

- "No change"

### CODE REVIEW

**Review by ricellis — APPROVED** [review]
I really like the proposed structure here, it makes a lot of sense.
I have two suggestions:
1. `connect/transforms` -> `transforms` (having seen all the other classes tidied up this extra package level seems unnecessary now)
1. `common/utils` -> `utils` (we could consilidate the constants and utils in a single package; there doesn't seem a lot of value in calling it `common` since virtually all the packages are common between source and sink anyway)

**Comment by mojito317** [comment]
> 1. `connect/transforms` -> `transforms` (having seen all the other classes tidied up this extra package level seems unnecessary now)

I have not removed the `connect` package yet, because that structure resembles the `org.apache.kafka.connect.transforms`. If that does not have any benefit, I can let it go and have just `transforms` without `connect`. Do you have anything in mind that is besides to keep `connect.transforms`?

**Comment by ricellis** [comment]
>If that does not have any benefit, I can let it go and have just transforms without connect. Do you have anything in mind that is besides to keep connect.transforms?

I think we either have the `connect` package as a "container" for all the other packages or we don't have it at all. Since all the code here is related to the Kafka connect API. I don't think we need an extra package level to indicate that really and AFAIK there is no good reason to have `connect.transforms` as opposed to just `transforms`. @tomblench any concerns?

**Comment by tomblench** [comment]
> > If that does not have any benefit, I can let it go and have just transforms without connect. Do you have anything in mind that is besides to keep connect.transforms?
> 
> I think we either have the `connect` package as a "container" for all the other packages or we don't have it at all since all the code here is related to the Kafka connect API. I don't think we need an extra package level to indicate that really and AFAIK there is no good reason to have `connect.transforms` as opposed to just `transforms`. @tomblench any concerns?

Agree that `connect.transforms` should become `transforms` and then I think we're all good.

File: README.md [hunk 1]
```diff
@@ -163,7 +163,7 @@ The examples below demonstrate modifying records produced by the Cloudant source
     transforms.omitDesignDocs.predicate=isDesignDoc
 
     predicates=isDesignDoc
-    predicates.isDesignDoc.type=com.ibm.cloud.cloudant.kafka.connect.transforms.predicates.IsDesignDocument
+    predicates.isDesignDoc.type=predicates.com.ibm.cloud.cloudant.kafka.transforms.IsDesignDocument
```
[the remark this answers is not in the document]
Comment (mojito317) [reply README.md]: Fixed in https://github.com/IBM/cloudant-kafka-connector/pull/115/commits/e83409a25c06f827a84207133973da25d3d13ed8

**Review by tomblench — APPROVED** [review]

**Pull request merged**